### ISSUE TITLE
Unexpected failure with def_property* on latest master

### ISSUE THREAD

**Issue by pschella** [pybind/pybind11#460]
The following code suddenly stopped working when upgrading to latest master:

`cls.def_property_readonly_static("dimensions", [](py::object /* self */) { return 3; });`

with error:

```include/pybind11/pybind11.h:1042:13: note: candidate function [with Extra = <>] not viable: no known conversion from '(lambda at python/lsst/afw/geom/point.cc:123:52)' to 'const pybind11::cpp_function' for 2nd argument
     class_ &def_property_readonly_static(const char *name, const cpp_function &fget, const Extra& ...extra) {
             ^```

any ideas?

**Comment by dean0x7d** [comment]
Accidentally broken after #449. Should be fixed with #458, but it's not in master yet.

**Comment by pschella** [comment]
Great, thanks!
